src/views/home/hooks/usePriceETH.ts

```diff
@@ -0,0 +1,64 @@
+import RToken from 'abis/RToken'
+import { ListedToken } from 'hooks/useTokenList'
+import { useMemo } from 'react'
+import { formatEther, getAddress } from 'viem'
+import { useContractReads } from 'wagmi'
+
+const usePriceETH = ({
+  id,
+  chain,
+  price,
+  supply,
+  targetUnits,
+  basketsNeeded,
+}: Partial<
+  Pick<ListedToken, 'id' | 'chain' | 'price' | 'supply' | 'targetUnits'> & {
+    basketsNeeded: number
+  }
+>) => {
+  const { data } = useContractReads({
+    contracts: [
+      ...(targetUnits === 'ETH' && id && chain && (!basketsNeeded || !supply)
+        ? [
+            {
+              address: getAddress(id),
+              abi: RToken,
+              functionName: 'basketsNeeded',
+              chainId: chain,
+            },
+            {
+              address: getAddress(id),
+              abi: RToken,
+              functionName: 'totalSupply',
+              chainId: chain,
+            },
+          ]
+        : []),
+    ],
+    allowFailure: false,
+  })
+
+  const basketNeededValue = useMemo(() => {
```

Comment:
Kind of overkill, I would do a memo for the whole `data` as that value is already memoized, then do all the calculations an return there.

```useMemo(..., [data])```

In this case thought, all the calculations are simple, even a memo feels weird